explorer/models/taskNode.ts

```diff
@@ -0,0 +1,66 @@
+import { ResourceManagementClient, SubscriptionClient, SubscriptionModels } from 'azure-arm-resource';
+import * as opn from 'opn';
+import * as vscode from 'vscode';
+import * as ContainerModels from '../../node_modules/azure-arm-containerregistry/lib/models';
+import { AzureAccount, AzureSession } from '../../typings/azure-account.api';
+import { AzureCredentialsManager } from '../../utils/azureCredentialsManager';
+import { NodeBase } from './nodeBase';
+
+/*Single TaskRootNode under each Repository. Labeled "Build Tasks" */
+export class TaskRootNode extends NodeBase {
+    constructor(
+        public readonly label: string,
+        public readonly contextValue: string,
+        public subscription: SubscriptionModels.Subscription,
+        public readonly azureAccount: AzureAccount,
+        public registry: ContainerModels.Registry,
+        public readonly iconPath: any = {}
+    ) {
+        super(label);
+    }
+
+    public name: string;
+
+    public getTreeItem(): vscode.TreeItem {
+        return {
+            label: this.label,
+            collapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+            contextValue: this.contextValue,
+            iconPath: this.iconPath
+        }
+    }
+
+    /*Making a list view of BuildTaskNodes, or the Build Tasks of the current registry */
+    public async getChildren(element: TaskRootNode): Promise<BuildTaskNode[]> {
+        const buildTaskNodes: BuildTaskNode[] = [];
+        let buildTasks: ContainerModels.BuildTask[] = [];
+
+        const client = AzureCredentialsManager.getInstance().getContainerRegistryManagementClient(element.subscription);
+        const resourceGroup: string = element.registry.id.slice(element.registry.id.search('resourceGroups/') + 'resourceGroups/'.length, element.registry.id.search('/providers/'));
```

Comment:
I remember I see in the previous PR the codebase has several places trying to parse the id string to get the resource group. Can you double check and move them to a shared utility module? If it doesn't fit in current PR, you can open an issue to track it.

Reply:
Do you mean making a function to parse the id in utils.ts? @northtyphoon

Reply:
Yes. If you see other codes do the same parsing, you should also update them to use the new method you will create.

Reply:
New function is called getResourceGroup( ) in acrTools.ts